src/Parser.php

```diff
@@ -438,16 +419,43 @@ protected function parseOutline()
             throw new UnexpectedParserNodeException('Step or Examples table', $node, $this->file);
         }
 
-        if (count($examples) === 0) {
-            throw new ParserException(sprintf(
-                'Outline should have examples table, but got none for outline "%s" on line: %d%s',
-                rtrim($title),
-                $line,
-                $this->file ? ' in file: ' . $this->file : ''
-            ));
+        if ($examples !== []) {
+            return new OutlineNode(rtrim($title) ?: null, $tags, $steps, $examples, $keyword, $line);
         }
 
-        return new OutlineNode(rtrim($title) ?: null, $tags, $steps, $examples, $keyword, $line);
+        return new ScenarioNode(rtrim($title) ?: null, $tags, $steps, $keyword, $line);
+    }
+
+    /**
+     * Peek ahead to find the node that the current tags belong to.
+     *
+     * @throws UnexpectedTaggedNodeException if there is not a taggable node
+     */
+    private function predictNextTaggedNodeType(): string
+    {
+        $deferred = [];
+        try {
+            while (true) {
+                $deferred[] = $next = $this->lexer->getAdvancedToken();
+
+                if (in_array($next['type'], ['Tag', 'Comment', 'Newline'], true)) {
```

Comment:
Nitpick: I would probably assign `$next['type']` to a variable instead of accessing it several times